tests/unit/s2n_self_talk_session_resumption_test.c

```diff
@@ -39,34 +129,36 @@ void mock_client(int writefd, int readfd)
     /* Give the server a chance to listen */
     sleep(1);
 
+    /* Initial handshake */
     conn = s2n_connection_new(S2N_CLIENT);
     config = s2n_config_new();
     s2n_config_disable_x509_verification(config);
     s2n_connection_set_config(conn, config);
-    conn->server_protocol_version = S2N_TLS12;
     conn->client_protocol_version = S2N_TLS12;
-    conn->actual_protocol_version = S2N_TLS12;
 
     s2n_connection_set_read_fd(conn, readfd);
     s2n_connection_set_write_fd(conn, writefd);
 
-    /* Set non-empty session id to indicate that we want to try to resume session */
-    memcpy(conn->session_id, SESSION_ID, S2N_TLS_SESSION_ID_MAX_LEN);
-    conn->session_id_len = S2N_TLS_SESSION_ID_MAX_LEN;
-
     if (s2n_negotiate(conn, &blocked) != 0) {
         result = 1;
     }
 
-    /* Now ensure that we got a new session id since server forced full handshake */
-    if (memcmp(conn->session_id, SESSION_ID, S2N_TLS_SESSION_ID_MAX_LEN) == 0) {
+    /* Make sure we did a full handshake */
+    if (!IS_FULL_HANDSHAKE(conn->handshake.handshake_type)) {
         result = 2;
     }
 
-    if (s2n_send(conn, MSG, sizeof(MSG), &blocked) != sizeof(MSG)) {
+    /* Save session state from the connection */
+    memset(serialized_session_state, 0, sizeof(serialized_session_state));
+    serialized_session_state_length = s2n_connection_get_session_length(conn);
+    if (s2n_connection_get_session(conn, serialized_session_state, serialized_session_state_length) != serialized_session_state_length) {
```

Comment:
We should check that length is below sizeof(serialized_session_state), before passing it into s2n_connection_get_session

Reply:
Done.